### ISSUE TITLE
Union attribute missing in response

### ISSUE THREAD

**Issue by macpoint** [goadesign/goa#3075]
I have design like this

```go
var Task = Type("Task", func() {
	Attribute("taskName", String, func() {
		MinLength(5)
	})
	Attribute("taskPayload", TaskPayload)
	Required("taskName", "taskPayload")
})

var TaskPayload = Type("TaskPayload", func() {
	OneOf("taskPayload", func() {
		Description("Specific task payload")
		Attribute("restartNodeTask", RestartNodeTask)
		Attribute("setDonorTask", SetNodeAsDonorTask)
	})
})

var SetNodeAsDonorTask = Type("setNodeAsDonorTask", func() {
	Description("Set node as donor")
	Attribute("cluster", String, func() {
		MinLength(5)
	})
	Attribute("donor", String, func() {
		MinLength(5)
	})

	Required("cluster", "donor")
})

// var RestartNodeTask similar to SetNodeAsDonorTask
```

```
$ goa gen ...
$ goa example ...
```

My method states just `Result(Task)` and `Payload(Task)`. 

The request:
```
{
  "taskName": "mytask",
  "taskPayload": {
    "Type": "setNodeAsDonorTask",
    "Value": {
      "cluster": "cluster1",
      "donor": "donor1"
    }
  }
}
```

But the `taskPayload` is always nil in the response:

```
{"taskName":"mytask","taskPayload":{}}
```
Am I missing something (like messing up with the Result option in my method)?

**Comment by raphael** [comment]
Are you able to post a complete example somewhere? 

One thing to check: the design defines an attribute `taskPayload` on the `TaskPayload` type so to initialize a value the code would have to do something like:
```go
p := &gensvc.TaskPayload{TaskPayload: gensvc.RestartNodeTask}
```
where `gensvc` is the generated package for the service using the type.

**Comment by macpoint** [comment]
Thanks @raphael for the response. 
I can initiate the value like so:

```go
// Create implements create.
func (s *taskssrvc) Create(ctx context.Context, p *tasks.Task) (res *tasks.Task, err error) {
	p.TaskPayload = &tasks.TaskPayload{TaskPayload: &tasks.RestartNodeTask{
		Cluster: "a",
		NodeName: "b",
	}}
	
	// panics as taskPayloadVal() is nil
	// restart := p.TaskPayload.TaskPayload.(*tasks.RestartNodeTask)

	res = p
	s.logger.Print("tasks.create")
	return
}
```

But how do I populate the `TaskPayload` with values from the request?

This is my method:

```go
var _ = Service("tasks", func() {
	Description("API for tasks")

	Method("create", func() {
		Payload(Task)
		Result(Task)
		HTTP(func() {
			POST("/tasks")
		})
	})

	Files("/openapi3.json", "./gen/http/openapi3.json")
})
```

The is request taskPayload is also nil in the response:
```
{
  "taskName":"mytaskww",
  "taskPayload": {
    "taskPayload": {
      "type":"restartNodeTask",
      "value":{
        "nodename":"a",
        "cluster":"b"
      }
    }
  }
}
```

It looks as if the contents of the `value` from the request is not processed at all.
The `type` is, however. This is an expected response for the incorrect task type:
```
{"name":"invalid_enum_value","id":"1YR-6Hqh","message":"value of body.taskPayload.Type must be one of \"restartNodeTask\", \"setNodeAsDonorTask\" but got value \"XrestartNodeTask\"","temporary":false,"timeout":false,"fault":false}
```

Similarly, expected response for the missing `value` key:

```
{"name":"missing_field","id":"aVLUkwHL","message":"\"Value\" is missing from body.taskPayload","temporary":false,"timeout":false,"fault":false}
```

**Comment by raphael** [comment]
Thank you for the report, this is a bug. I have a tentative fix in the branch `fix_union_http`, would you mind trying it out:
```bash
go get goa.design/goa/v3@fix_union_http
goa gen ...
```

**Comment by raphael** [comment]
@macpoint are you able to confirm this is fixed in v3.7.10? thanks!
